### PR TITLE
Add StandardRange::size method and use to get node degree.

### PR DESCRIPTION
**Pull request by arthurp**
Based on idea discussed here: https://github.com/KatanaGraph/katana/pull/35#issuecomment-761578291

Also fixes a stray bug.

After: #31 (simply to avoid a complex rebase)

### CODE REVIEW

File: libgalois/src/analytics/Utils.cpp [hunk 1]
```diff
@@ -19,15 +19,15 @@
...
 
 uint32_t
 katana::analytics::SourcePicker::PickNext() {
   uint32_t source;
   do {
     source = RandomUniformInt(graph.size());
-  } while (
-      std::distance(graph.edges(source).begin(), graph.edges(source).end()));
+  } while (graph.edges(source).empty());
+  KATANA_LOG_ASSERT(graph.edges(source));
```
Comment (roshandathathri) [inline libgalois/src/analytics/Utils.cpp]: Confused by this. `edges()` returns a range, so what is this asserting?

Also, I think `assert()` (debug-only) is better here than `KATANA_LOG_ASSERT` (which is also invoked in release builds).

**Review by roshandathathri — APPROVED** [review]
Yay. LGTM

File: libgalois/src/analytics/Utils.cpp (hunk 1 above)
Comment (arthurp) [reply libgalois/src/analytics/Utils.cpp]: C++ ranges are "true-ish" if they are not empty. So this is equivalent to `graph.edges(source).size() > 0`.

Yea. Debug only would be good. But I'd like to have a KATANA macro for it. The reason is that the aborting KATANA macros will eventually actually throw an exception to abort the C++ code in a way that doesn't crash the Python interpreter. Python extensions are not supposed to abort. I'll post about this on code-related.

Comment (roshandathathri) [reply libgalois/src/analytics/Utils.cpp]: Thanks

**Comment by arthurp** [comment]
@roshandathathri I had to fix another bug (the last commit). Do you want to check it?

**Review by roshandathathri — APPROVED** [review]
LGTM

**Pull request merged**